lib/puppet_x/puppetlabs/property/region.rb

```diff
@@ -0,0 +1,15 @@
+module PuppetX
+  module Property
+    class AwsRegion < Puppet::Property
+      validate do |value|
+        name = resource[:name]
+        fail 'region should not contain spaces' if value =~ /\s/
+        fail 'region should not be blank' if value == ''
+        fail 'region should be a String' unless value.is_a?(String)
+        if !ENV['AWS_REGION'].nil? && ENV['AWS_REGION'] != value
+          fail "if using AWS_REGION environment variable it must match the specified region value for #{name}"
```

Comment:
For my own understanding, this is the case because we call *_client methods in the providers, and those use the default region, correct?